core/modules/replica/Configuration.h

```diff
@@ -429,75 +430,610 @@ class Configuration: public ConfigurationIFace {
+     * be applying to the internal state.
+     * @param lock The lock on '_mtx' to be acquired prior to calling the method.
+     * @param obj An object to be used as a source for amending th edefault values
+     *   of the configuration parameters. Note that this source doesn't assume a presence
+     *   of any persistent back-end for the configuration. If any such back-end existed
+     *   then it would be disconnected upon the completion of the method.
+     * @param reset The flag (if set to 'true') will trigger the internal state reset
+     *   to the default values of the parameters before applying the input configuration.
+     */
+    void _load(util::Lock const& lock, nlohmann::json const& obj, bool reset);
+
+    /**
+     * Load from MySQL. Parameters read from the database will be applied
+     * to the internal state.
+     * @param lock The lock on '_mtx' to be acquired prior to calling the method.
+     * @param configUrl The connection string to the MySQL database from which to read
+     *   the configuration parameters. The database will become the persistent back-end
+     *   for the configuration.
+     * @param reset The flag (if set to 'true') will trigger the internal state reset
+     *   to the default values of the parameters before applying the input configuration.
+     */
+    void _load(util::Lock const& lock, std::string const& configUrl, bool reset);
+
+    /// @param lock The lock on '_mtx' to be acquired prior to calling the method.
+    /// @param showPassword If a value of the flag is 'false' then hash a password in the result.
+    /// @return The JSON representation of the object.
+    nlohmann::json _toJson(util::Lock const& lock, bool showPassword=false) const;
+
+    /**
+     * Return a JSON object representing a parameter.
+     * @param lock The lock on '_mtx' to be acquired prior to calling the method.
+     * @param category The name of the parameter's category.
+     * @param param The name of the parameter within its category.
+     * @return A 'const' reference to a JSON object encapsulating the parameter's value and its type.
+     * @throws std::invalid_argument If the parameter doesn't exist.
+     */
+    nlohmann::json const& _get(util::Lock const& lock,
+                               std::string const& category,
+                               std::string const& param) const;
+
+    /**
+     * Return a modifiable JSON object representing a parameter.
+     * @note The parameter is not required to exist at a time of the call. It would be
+     *   created automatically with no specific value or type set.
+     * @param lock The lock on '_mtx' to be acquired prior to calling the method.
+     * @param category The name of the parameter's category.
+     * @param param The name of the parameter within its category.
+     * @return A reference to a JSON object encapsulating the parameter's value and its type.
```

Comment:
I didn't get the question. If you look at the documentation on the JSON library at https://github.com/nlohmann/json then it will be clear that unlike the string-based key/val maps the JSON structures are type-safe since they store values of the final types, not strings. There is no dynamic type "conversion" per-se. There is run-time type-checking and typecasting if the proposed type is right. Otherwise, an exception will be thrown.

Reply:
The `@note` above says:

    @note The parameter is not required to exist at a time of the call. It would be
     *   created automatically with no specific value or type set.